api/main.py

```
@@ -32,6 +64,97 @@
+
+        # Save to Supabase
+        dummy_user_id = os.getenv("DUMMY_USER_ID")
+        if dummy_user_id:
+            try:
+                supabase.table('history').insert({
+                    'original_text': text_input.text,
+                    'improved_text': improved_text,
+                    'user_id': dummy_user_id
+                }).execute()
+                print(f"Saved history for user {dummy_user_id}")
+            except Exception as e:
```

Ruff (0.12.2)

Do not catch blind exception: `Exception`

(BLE001)